### DISCUSSION TITLE
全局配置适配器类型

### DISCUSSION THREAD

**Discussion by yuhengAlioth** [Zhengqbbb/cz-git#170] [Q&A]
我是windows系统，使用了nvm，官方文档说看一下系统环境变量有没有路径，我看了没有，但是我不知道怎么去配置让其生效，求解答

**Comment by Zhengqbbb** [comment]
> 我是windows系统，使用了nvm，官方文档说看一下系统环境变量有没有路径，我看了没有，但是我不知道怎么去配置让其生效，求解答

1. 什么报错？命令找不到吗？
2. 什么 node 版本？
3. 使用了 wsl 吗？还是普通的 CMD
4. 只有  `czg` 命令还是说其他全局安装的软件包，命令都找不到

---

ref https://github.com/Zhengqbbb/cz-git/discussions/155#discussioncomment-8301732

> 其实一个命令之所以可以在终端通过名字的方式直接调用，关键就是在这些包的目录有没有在全局的 $PATH 之中
类似这个：https://juejin.cn/post/6946594249093677069#heading-6 。在 Windows 中设置 $PATH 没办法通过命令的方式，或者说没右键设置这么简单。我记得以前在 Windows 中设置 JAVA_HOME 总是很烦

**Reply by yuhengAlioth** [reply]
作者大佬，是这样的我最开始安装的是node18，因为其他项目所以安装了nvm并且安装了node16，现在我使用18安装了全局之后，在全局配置适配器类型这一步运行代码说系统找不到指定的路径。安装了wsl但是没用，只是使用普通cmd。

**Reply by Zhengqbbb** [reply]
1. 也就是说你 **切换** 你的版本后，全局安装 `npm install` 后发生了找不到命令 
    nvm 虽然帮你切换了 node 的版本，但是没有帮你把 node18 全局安装包路径设置在  **系统环境变量之中**

---
> 小科普：一个命令之所以在终端直接打命令名就可以启动，是因为命令可执行文件存放目录设置在 **系统环境变量之中** 也就是 `$PATH` 之中
1. 先查看你当前环境中的 npm 包下载**目录** ：`npm prefix -g`
2. 查看 Windows 系统环境变量中是否有这个目录：没有就添加咯，有就看看目录里面有没有 `czg` 这个文件
   - 查看：我的电脑 -> 属性 -> 高级系统设置 -> 环境变量 -> 系统变量 -> path
   - 添加：此电脑 -> 属性 -> 高级系统设置 -> 环境变量 -> 系统变量 -> path ->编辑 - > 新增路径

> 小科普：nvm 为什么有这样做，是因为 node 版本控制器都具有版本隔离机制，以防止不同版本造成原先可运行的软件包无法使用

<img width="1186" alt="image" src="https://github.com/Zhengqbbb/cz-git/assets/40693636/6a4ca2ab-bbd8-4586-af51-b1f8fe9814b0">

**Reply by yuhengAlioth** [reply]
作者大大，我解决这个问题了，不用去配置环境变量，只需要全局安装之后，在 C盘 -> User -> 登录用户 里面直接创建.czrc文件，里面使用JSON 模板即可，这样不管那个版本的node都可以使用，但是得使用git cz这个命令提交就可以啦

**Reply by Zhengqbbb** [reply]
原来不是命令找不到，是运行报错了，而且使用的还是 `commitzen` CLI，我以为你使用的是 `czg` CLI，如果你贴了报错信息就好了，这样一看就知道了

`commitzen` CLI 是要配置指明你要使用的适配器 `.czrc`<sup>即使用什么 UI 插件</sup>
且 .czrc 是要放在 windows 之中磁盘的 Node 所在的 HOME 目录：
即 如果你 Node 安装在 C 盘就在 C 盘登录用户目录之中创建

祝你使用愉快啦
